### PR TITLE
[tests] Hopefully this will simplify the test suite (work in progress)

### PR DESCRIPTION
**Pull request by jgsogo**

### CODE REVIEW

**Review by uilianries — COMMENTED** [review]
Still in draft, but looks promising. Clean and professional with PyTest, I really like it.

File: modern/_tests/workingdir/simple/test_simple.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (uilianries) [inline modern/_tests/workingdir/simple/test_simple.sh]: I wanted to move all as Python scripts, including those test_xxx.sh files. When I started this new feature, it was all bash script, but when I opened the PR, I converted some of them to Python. WDYT, should we convert all to Python?

Comment (jgsogo) [reply modern/_tests/workingdir/simple/test_simple.sh]: I would say that we need them as `.sh` script. With _bash_ we can run them in _all_ the images, there is no need to install python. For example, we could run these tests in `builder` images.

File: modern/docker-compose.yml [hunk 2]
```diff
@@ -5,11 +5,10 @@ x-common-args: &common-args
     CMAKE_VERSION: ${CMAKE_VERSION}
     PYTHON_VERSION: ${PYTHON_VERSION}
     CONAN_VERSION: ${CONAN_VERSION}
-    DOCKER_USERNAME: ${DOCKER_USERNAME}
     LIBSTDCPP_VERSION: ${LIBSTDCPP_VERSION}
-    LIBSTDCPP_PATCH_VERSION: ${LIBSTDCPP_PATCH_VERSION}
-    ARTIFACTORY_REPOSITORY: ${ARTIFACTORY_REPOSITORY}
-    JENKINS_AGENT_VERSION: ${JENKINS_AGENT_VERSION}
+    DOCKER_USERNAME: ${DOCKER_USERNAME}
+    ARTIFACTORY_REPOSITORY: ${ARTIFACTORY_REPOSITORY:-https://c3istg.jfrog.io/artifactory/dad-generic}
```
Comment (uilianries) [inline modern/docker-compose.yml:11]: cool, didn't know about default values.

**Comment by jgsogo** [comment]
No longer useful.

**Pull request closed without merging**